Feature/t3d pipeline package

TomSimons opened the pull request.

TomSimons committed:
Added folder structure and script to request CityJSON from a BAG ID

TomSimons committed:
Added events as described in the package guidelines, still kept option for direct interaction. Added sample scene

TomSimons committed:
Started adding CityJSON parser

TomSimons committed:
Finished basic parser and exporter. still need to add semantics in geometry

TomSimons committed:
Added Parsing semantics and exporting them again

TomSimons committed:
Added semantic null propagation

TomSimons committed:
Removed comments, checked if custom types are processed correctly

TomSimons committed:
Added comments for null propagation of Semantics Values

TomSimons committed:
Started adding CityJSON visualizer

TomSimons committed:
Added default material, fixed mesh direction, removed vertexOffset

TomSimons committed:
added serialized option to use json center as relative RD center

TomSimons committed:
removed prints

TomSimons committed:
Added vertex transform processing

TomSimons committed:
Removed Write Text File script

TomSimons committed:
cleanup

TomSimons committed:
Removed duplicate vertices from the export

TomSimons committed:
Fixed vertex indexing, added geographical extents to metadata node

TomSimons committed:
Started adding annotation functionality

TomSimons committed:
Annotations can be typed and are added to the CityJSON Export

TomSimons committed:
added option for local/global id

TomSimons committed:
Added annotation markers and tooltips

TomSimons committed:
Added upload scripts and assets

TomSimons committed:
Cleanup

TomSimons committed:
Fixed hierarchy to allow for uploading exports. Authentication token not included

TomSimons committed:
Started adding comments

TomSimons committed:
Added missing comments, minor fixes

TomSimons committed:
Fixed sample scenes

sambaas reviewed (changes requested):
- De samples map is nog niet importeerbaar in de Package Manager, en is zichtbaar nog zichtbaar in Unity in de package structuur (geen ~ als suffix) dus heb de samples scene nog niet kunnen testen.

- Er bestaat al een Poly2Mesh in het project, die valt nu onder de selectiontools. 
Is het een idee om dit pakket en de selectiontools een losse dependecy te geven op een 'MeshManipulation' package, of iets in die trend? De Poly2Mesh library is nagenoeg hetzelfde, alleen hebben we nog een toevoeging gedaan zodat je ook polygonen een dikte / extrude kan geven

TomSimons committed:
Separated Poly2Mesh into a separate dependency in ThirdPartyTools. T3DPipeline and SelectionTools now depend on this. Used version from SelectionTools because this version had some additions.

TomSimons committed:
- A CityJSON with an empty vertex list will now no longer throw an error. it will give a warning that any visualisation will create empty meshes.
- Removed deleted script from Scene
- added todo for MultiPoint and MultiLineString mesh creation

TomSimons commented:
Punt 2 is aangepast, punt 1 lukt niet bij mij, hij verwijdert de hele samples map dan in git.

sambaas committed:
Merge branch 'main' of https://github.com/Amsterdam/Netherlands3D into feature/T3DPipeline-package

sambaas committed:
moved T3D samples into samples~

TomSimons committed:
added samples path to package.json

TomSimons committed:
Fixed missing reference
